### ISSUE TITLE
[Bug]: SSgetoutput and SS_output throw an error when the covar.ss file contains NAs

### ISSUE THREAD

**Issue by michellesculley** [r4ss/r4ss#830]
### Describe the bug

I'm not sure if this is technically a bug, but an unhelpful error. If you have a model run that the covar.sso file is created and contains some data, but some parameters have NA for the standard deviations, the SS_output and SSgetoutput commands will through an error and not load the model. The Error is: Error in if (Nstd <= 1) { : missing value where TRUE/FALSE needed. When running SSgetoutput, the code stops at whichever model is giving the error and doesn't load the rest of the list of models.

### To Reproduce

1. Go to your covar.sso file, change one of the parameter standard deviations to nan.
2. Try to load the model into R with r4ss.

### Expected behavior

I would expect SSgetoutput to give the message "!repfile doesn't exists or is empty" and then continue loading the next model on the list as it does when there are other problems loading a model. 

### Which versions of r4ss and SS3 are you seeing the problem on?

r4ss version 1.47.0, SS3 V3.30.18.00

### Additional Context

Generally rerunning the model produces a covariance file without NA, but I would not have known that was the problem without digging into the source code of r4ss. A more informative error like the one given when Nstd<=1 is TRUE (Too few estimated quantities in covar file (n=", Nstd, "). Change input to covar=FALSE.) would be nice.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/r4ss/r4ss#code-of-conduct)

**Comment by iantaylor-NOAA** [comment]
Thanks @michellesculley for reporting this. I just pushed a fix in commit f1ac587 where it nows plays through with a warning like this
```
Warning message:
In SS_output(dir = mydir, repfile = repFileName, covarfile = covarname,  :
  NA value for parameter uncertainty found in 1 rows of covar.sso file. First par with NA: NatM_uniform_Fem_GP_1  
```
The `SSgetoutput()` function will keep going when this happens to one of the models, but it won't be clear which of the models had the issue, so you would have to track that down further which model had the problem. That is, the warning appears only after the function is run. I think that's similar to most of the other warnings that you would get when running `SS_output()`.

Let me know if this seems adequate or if there are further refinements that you would like.
If you have time, please test the change by installing via `pak::pkg_install("r4ss/r4ss@NA_in_covar_issue_830")`. Otherwise I will merge into the main branch tomorrow.

Bigger question: do you know why there is an NA value in covar.sso and should we look into what could be causing that?

**Comment by michellesculley** [comment]
Thanks. I think that's fine. It works for me.

I'm not sure why there are NA's in the covar file. I've noticed it has started to happen with SS3.30.18 and later versions, I think it might have more to do with the most recent updates to admb, if I had to guess. It is usually just a few of the parameters that are a problem, most have estimated values as you would expect. Generally re-running the model solves the problem, but you don't get a warning that the hessian isn't positive definite or anything, which was a bit surprising as I would have expected a problem there if there were NAs in the covariances.

Thanks for your quick response!

**Comment by Rick-Methot-NOAA** [comment]
Glad to take a look, but may be a needle in haystack.  Is it always the same parameter?  Is it a covar line, or a parm_std line?

**Comment by michellesculley** [comment]
Yeah, unfortunately its not very reproducible, as rerunning the model usually solves the problem. I have noticed that it is typically one or more derived parameters, not estimated parameters, and for the parameter(s) that have nan for the value, it is the standard deviation and all the covariances with that parameter.

**Comment by Rick-Methot-NOAA** [comment]
I suspect something is causing a divide by 0 in this function in ss_objfun.tpl
FUNCTION void Process_STDquant()

There are many division statements in that code

**Comment by iantaylor-NOAA** [comment]
The r4ss part of this issue was resolved by Pull Request #832.
If/when we find a model where this occurs we can open an issue in the stock-synthesis repository to debug the root cause.
